### PR TITLE
finished assessment

### PR DESCRIPTION
**Pull request by justinhilario**

### CODE REVIEW

File: code-challenges.test.js [hunk 1]
```diff
@@ -22,30 +22,53 @@ var secretCodeWord1 = "lackadaisical"
...
+    })
+})
 
 
 // b) Create the function that makes the test pass.
 
-
-
+const enigmaMachine = (string) => {
+    let coded = string.replace(/e/ig,'3').replace(/a/ig,'4').replace(/i/g,'1').replace(/o/g,'0')
```
Comment (minachuong) [inline code-challenges.test.js:42]: good logic. Consider a way to organize the mapping from letter to number without having to iterate on the string multiple times.

File: code-challenges.test.js [hunk 2]
```diff
@@ -22,30 +22,53 @@ var secretCodeWord1 = "lackadaisical"
...
+        expect (findsTheA(arrayOfWords)).toEqual(wordsWithA)
+    })
+})
 
 
 
 // b) Create the function that makes the test pass.
 
-
+const findsTheA = () => {
```
Comment (minachuong) [inline code-challenges.test.js:66]: 👍

File: code-challenges.test.js [hunk 3]
```diff
@@ -61,7 +84,45 @@ var hand3 = [5, 5, 5, 5, 4]
...
+})
+
 
 
 
 // b) Create the function that makes the test pass.
+
+
+const fullHouse = (hand) => {
+    for(let i=0; i<hand.length; i++){
```
Comment (minachuong) [inline code-challenges.test.js:112]: This is definitely one way to check for some of the test cases, but this logic is dependent on values within the array having a certain order. For instance, an array like [2,3,2,3,3] would return false but it is a full house

Comment (minachuong) [reply code-challenges.test.js:112]: also it seems that the for-loop isn't necessary because the logic within the for-loop isn't dependent on the iteration or accessing the array using an index

File: interview-questions.md [hunk 4]
```diff
@@ -7,64 +7,68 @@ Then, research the question to expand on your answer. Even if you feel you have
...
 
-  Researched answer:
+  Researched answer: CRUD is an acronym for the four basic types of SQL commands: Create , Read , Update , Delete. A CRUD application is one that uses forms to get data into and out of a database.
 
 
 
 3. What is a relational database? Are there other kinds of databases?
 
-  Your answer:
+  Your answer: Relational databases are able to store data in tables, and can form relationships between multiple tables.
```
Comment (minachuong) [inline interview-questions.md:26]: 💯

File: interview-questions.md [hunk 5]
```diff
@@ -7,64 +7,68 @@ Then, research the question to expand on your answer. Even if you feel you have
...
 
-  Researched answer:
+  Researched answer: A relational database is a collection of data items with pre-defined relationships between them. These items are organized as a set of tables with columns and rows. Tables are used to hold information about the objects to be represented in the database.
 
 
 
 4. What are the 5 HTTP verbs? What are they important?
 
-  Your answer:
+  Your answer: Post, get, put, patch, and delete. They correspond to each CRUD function for HTTP based web clients.
```
Comment (minachuong) [inline interview-questions.md:34]: 💯

File: rails-questions.md [hunk 6]
```diff
@@ -2,22 +2,36 @@
 
 1. What is the command to create a new Rails application with a Postgres database?
 
+First we need to create a new rails app ->
 
+$ rails new rails_app -d postgresql -T
```
Comment (minachuong) [inline rails-questions.md:7]: 💯

File: rails-questions.md [hunk 7]
```diff
@@ -2,22 +2,36 @@
...
 
+Foreign keys are done to add an association between two tables. To make an association between table1 and table2, we assign the foreign key to table2.
+In this case, the foreign key is a column in table2 called table1_id.
 
+To create the association between the Person and the Email table we had to assign a foreign key to Email. The foreign key always goes on the belongs_to side. By convention the foreign key should always be the modelname_id and the data type should be integer. The has_many model is singular, the belongs_to model is plural. A Person (singular) has_many emails (plural).
 
 3. What is the command to generate a Rails model for a person table with columns named shirt, pants, and shoes?
 
-
+$ rails generate model Person shirt:string pants:string shoes:string
```
Comment (minachuong) [inline rails-questions.md:21]: 💯

File: rails-questions.md [hunk 8]
```diff
@@ -2,22 +2,36 @@
...
 
 4. What is the naming convention for generating a Rails model? What is the naming convention for the table that is generated with this command?
 
+Model class - singular and PascalCase
 
+Database table - plural and snake_case
 
 5. A Rails application is going to have two models. One is called Library and one is called Book. A library will have many books and a book will belong to a library. Which model would contain the foreign key? What would the foreign key column be named?
 
+The Book model will contain the foreign key. and the foreign key column name would be library_id.
```
Comment (minachuong) [inline rails-questions.md:31]: 💯

**Review by minachuong — APPROVED** [review]
Great pseudocoding and code. Consider some edge cases outside of the ones provided to test for the "fidelity" of the function. Interview questions and rails questions are thorough 💯 !
